### PR TITLE
Revise how we set tiering and pgo options for spmi benchmark collections

### LINKED ISSUE #86410: revise how we pass config settings to benchmarkdotnet for SPMI collections
Currently we set some env vars via command line and inherit others from the environment.

This means the build and BDN host process are subject to some of the config settings, which isn't a problem but also isn't necessary or useful.

Eg for the benchmarks_pgo run, we see the actual benchmark process invocation is:
```
/datadisks/disk1/work/A7CB0920/p/superpmi/corerun --partition-count 30 --partition-index 0 --envVars DOTNET_JitName:$JitName  DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput
```
(note no mention of `DOTNET_TieredPGO` in the above. Instead it is set in the ambient environment:
```
[20:26:36] Starting collection using command
[20:26:36] export SuperPMIShimLogPath=/datadisks/disk1/work/A7CB0920/t/tmpv0oc5gd6
[20:26:36] export SuperPMIShimPath=/datadisks/disk1/work/A7CB0920/p/superpmi/libclrjit.so
[20:26:36] export DOTNET_EnableExtraSuperPmiQueries=1
[20:26:36] export DOTNET_TieredCompilation=1
[20:26:36] export DOTNET_TieredPGO=1
[20:26:36] export DOTNET_JitPath=/datadisks/disk1/work/A7CB0920/p/superpmi/libsuperpmi-shim-collector.so
[20:26:36] export DOTNET_JitName=libsuperpmi-shim-collector.so
[20:26:36] Collecting using command:
[20:26:36]   /datadisks/disk1/work/A7CB0920/t/tmpo5j9iv_h/run_microbenchmarks.sh 
....

Also for PGO/Tiering we really should be running more iterations and doing normal warmup.

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Don't set these in the BDN environment; set them via the BDN command line so they only impact the process being benchmarked.

Fixes #86410

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@kunalspathak PTAL
cc @dotnet/jit-contrib 

Internal pipeline run with this new script: green save unrelated OSX failure.

https://dev.azure.com/dnceng/internal/_build/results?buildId=2196191&view=results

File: src/coreclr/scripts/superpmi_benchmarks.py [hunk 1]
```diff
@@ -209,36 +209,48 @@ def build_and_run(coreclr_args, output_mch_name):
...
     else:
         # Disable ReadyToRun so we always JIT R2R methods and collect them
-        collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} --envVars DOTNET_JitName:{shim_name} " \
-                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
-                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+        collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} " \
+                              "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput " \
+                             f"--envVars DOTNET_JitName:{shim_name} DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0"
+
+    if coreclr_args.tiered_pgo:
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi_benchmarks.py:229]: do you mind also pulling the ` --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart ` in common case?

Also, shouldn't you be passing `iterationCount`, etc. if for tiered/pgo runs? And a note about `DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0` too. Do we still want them for tiered/pgo runs?

**Review by kunalspathak — CHANGES_REQUESTED** [review]

File: src/coreclr/scripts/superpmi_benchmarks.py (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/scripts/superpmi_benchmarks.py:229]: There is no common case right now, are you saying I should factor the common aspects out (or at least the common trailing elements)?

I wasn't planning on changing how we collect at this point. I would have to study the impact of altering BDN strategy to see if it adds value.

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_benchmarks.py:229]: > There is no common case right now, are you saying I should factor the common aspects out (or at least the common trailing elements)?

Yes, that's what I meant.

> I wasn't planning on changing how we collect at this point. I would have to study the impact of altering BDN strategy to see if it adds value.

Sure, we can limit this PR for just the changes you have currently and rethink about longer term strategy of how we should collect them in a later PR.

**Comment by kunalspathak** [comment]
> https://github.com/dotnet/runtime/pull/87082#discussion_r1215607612

Typically to test out things, I comment out all the unrelated collections in yml file and that way we just do the collection for type we are testing for. E.g. https://dev.azure.com/dnceng/internal/_build/results?buildId=2193577&view=results that just ran real-world collection.

**Comment by AndyAyersMS** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2196878&view=results is the revised run -- benchmark runs were all happy, though a test build failed (will restart it once I am able).

**Review by kunalspathak — APPROVED** [review]

**Review by EgorBo — APPROVED** [review]

**Pull request merged**